magefiles/config.go

```diff
@@ -0,0 +1,135 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"sort"
+	"strings"
+
+	"github.com/magefile/mage/mg"
+	"github.com/magefile/mage/sh"
+	"gopkg.in/yaml.v2"
+)
+
+// Content loaded from settings.yaml or settings.local.yaml
+// or file passed via SWDT_SETTINGS_FILE environment variable.
+var Settings map[string]string
+
+type Config mg.Namespace
+
+// Load configuration settings from default settings.yaml or user-specific settings.local.yaml.
+// First, checks for variables file set via SWDT_SETTINGS_FILE environment variables,
+// then user-specific settings.local.yaml, falling back to default varaibles.yaml.
+// Every public target must depend on Configure target.
+func (Config) Settings() error {
+	variablesFile, err := findVariablesFile()
+	if err != nil {
+		return err
+	}
+	variablesData, err := ioutil.ReadFile(variablesFile)
+	if err != nil {
+		return err
+	}
+	settings := make(map[string]string)
```

Comment:
Should we consider using a settings struct that you can unmarshal into?
```go
type settings struct {
    BuildFromSource bool `yaml:"build_from_source"`
    K8sLinuxKubeletNodeIP string `yaml:"k8s_linux_kubelet_nodeip"`
...
}
```
This way we don't need to keep a bunch of strings scattered across the code but can rather say `settings.BuildFromSource`.

Reply:
I agree, this is a valid point, from the engineering point of view.
But, I'm a Go _complete newbie_, so I wanted to keep things simple - after all, does it matter if I have to know what is `BuildFromSource` instead `"build_from_source"`. Using plain dictionary, for now, also allows me to fiddle with new properties, like just added `vagrant_vbguest_auto_update`, in one place of definition, instead of two.

Reply:
>  after all, does it matter if I have to know what is BuildFromSource instead "build_from_source"

@Aravindh Ha! I've just realised the power of the unmarshalling after I learned about field tags. I see your suggestion in new way now, awesome! I'll turned the map into bespoke type.

Reply:
@aravindhp We've got the `type Settings struct` now. This Go idiomatic approach gives lots of validation out of box!